contracts/adapters/HegicAdapter.sol

```diff
@@ -352,19 +352,55 @@ contract HegicAdapter is IProtocolAdapter {
     }
 
     /**
-     * @notice Function to bulk claim rHEGIC2 rewards from liquidity utilization
+     * @notice Function to get rHEGIC2 rewards claimable from liquidity utilization
+     * @param rewardsAddress is the address of the rewards contract (either for eth or wbtc)
+     * @param optionIDs is an array of all the option ids we want to claim rewards for
+     */
+    function rewardsClaimable(address rewardsAddress, uint256[] calldata optionIDs)
+        external
+        view
+        returns (uint256 rewardsAmount)
+    {
+        IHegicRewards rewardsContract = IHegicRewards(rewardsAddress);
+
+        uint256 i = 0;
+
+        while (i < optionIDs.length && optionIDs[i] > 0) {
+          (, , , uint _amount, , uint _premium, , ) = rewardsContract.hegicOptions().options(optionIDs[i]);
+          if(!rewardsContract.rewardedOptions(optionIDs[i])){
+            rewardsAmount = rewardsAmount.add(_amount.div(100).add(_premium)
+            .mul(rewardsContract.rewardsRate())
```

Comment:
Since `rewardsContract.rewardsRate` loads from storage, it will cost [200 gas per invocation](https://github.com/djrtwo/evm-opcode-gas-costs/blob/master/opcode-gas-costs_EIP-150_revision-1e18248_2017-04-12.csv).

Let's optimize this by loading it into a variable on top of the loop:

```solidity
uint rewardsRate = rewardsContract.rewardsRate();

while (...) {
  rewardsAmount = rewardsAmount.add(_amount.div(100).add(_premium)
             .mul(rewardsRate).div(1e8));
}
```